codex/sales.nim

```diff
@@ -139,21 +194,23 @@ proc mySlots*(sales: Sales): Future[seq[Slot]] {.async.} =
   return slots
 
 proc load*(sales: Sales) {.async.} =
-  let slots = await sales.mySlots()
+  let activeSlots = await sales.mySlots()
+
+  await sales.deleteInactiveReservations(activeSlots)
 
-  for slot in slots:
+  for slot in activeSlots:
     let agent = newSalesAgent(
       sales.context,
       slot.request.id,
       slot.slotIndex,
       some slot.request)
 
-    agent.context.onCleanUp = proc {.async.} = await sales.remove(agent)
+    agent.onCleanUp = proc {.async.} = await sales.remove(agent)
```

Comment:
Good catch. I must've missed this because of the rebased changes (on top of your changes that modified at which point we complete the processing future).

The loaded slots are not processed in the context of the slot queue, however, I've created a dummy future to pass in to `sales.cleanUp`.